### PR TITLE
Validation of property string patterns prior to trying to construct a URL

### PR DESCRIPTION
**Pull request by Akyiaa**
Signed-off-by: Fiona Ampofo <64271621+Akyiaa@users.noreply.github.com>
Story: https://github.com/galasa-dev/projectmanagement/issues/1607

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

File: docs/generated/errors-list.md [hunk 1]
```diff
@@ -130,8 +130,15 @@ The `galasactl` tool can generate the following errors:
 - GAL1134E: The runs reset operation failed. Unable to process the error information returned from the server.
 - GAL1135E: Error cancelling run '{}'. Reason: '{}'
 - GAL1136E: The runs cancel operation failed. Unable to process the error information returned from the server.
+<<<<<<< HEAD
+- GAL1137E: Unable to compile the regex pattern for Galasa Property field 'namespace'. Reason: '{}'
```
Comment (KirbyKatcher) [inline docs/generated/errors-list.md:134]: Can you please update the errors to prevent the conflict. running a build locally should fix this

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-216-l28nc for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/properties/propertiesSet_test.go [hunk 2]
```diff
@@ -261,3 +261,29 @@ func TestCreateGalasaPropertyReturnsOk(t *testing.T) {
 	assert.Equal(t, property.Metadata.GetName(), name)
 	assert.Equal(t, property.Data.GetValue(), value)
 }
+
+func TestInvalidNamesapceFormatReturnsError(t *testing.T) {
```
Comment (KirbyKatcher) [inline pkg/properties/propertiesSet_test.go:265]: Namesapce?

File: pkg/properties/propertiesSet_test.go [hunk 3]
```diff
@@ -169,7 +169,7 @@ func TestUpdatePropertyWithValidNamespaceAndVaidNameValueReturnsOk(t *testing.T)
 
 func TestUpdatePropertyWithInvalidNamesapceAndValidNameReturnsError(t *testing.T) {
 	//Given...
-	namespace := "invalidNamespace"
+	namespace := "invalidnamespace"
```
Comment (KirbyKatcher) [inline pkg/properties/propertiesSet_test.go:172]: why would this be invalid?

**Review by KirbyKatcher — CHANGES_REQUESTED** [review]

File: pkg/properties/propertiesSet_test.go (hunk 3 above)
Comment (Akyiaa) [reply pkg/properties/propertiesSet_test.go:172]: I see the issue, I have changed it to have a special character

**Comment by galasa-team** [comment]
Build successful

**Review by KirbyKatcher — APPROVED** [review]

**Pull request merged**